packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -254,6 +256,11 @@ export class CdkToolkit {
           }
 
           const confirmed = await promptly.confirm('Do you wish to deploy these changes (y/n)?');
+
+          setLogLock(false);
+          // Calls the logger and prints any logs that happened during the lock
+          print('');
```

Comment:
`setLogLock(false)` could/should flush buffered messages straight away, so you don't need the extra `print`.